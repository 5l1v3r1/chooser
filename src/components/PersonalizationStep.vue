<template>
        <section class="pers-form">
            <b-field label="Work Author" label-position="inside">
                <b-input
                    v-model="value.authorName"
                    placeholder="John Silver"
                />
            </b-field>
            <b-field
                label="Title of Work" label-position="inside"
              >
                <b-input
                    v-model="value.workTitle"
                    placeholder="This work"
                />
            </b-field>
            <b-field label="Work URL" label-position="inside">
                <b-input
                    v-model="value.workUrl"
                    placeholder="https://www.example.com/image.jpg"
                />
            </b-field>
            <b-field label="URL of the Source Work" label-position="inside">
                <b-input
                    v-model="value.sourceWorkUrl"
                    placeholder="https://www.example.com/source.jpg"
                />
            </b-field>
        </section>
</template>

<script>
export default {
    name: 'PersonalizationStep',
    props: ['value'],
    watch: {
        value() { this.$emit('input', this.$props.value) }
    }
}
</script>

<style lang="scss">
label.label {
    font-size: 0.845rem;
}
</style>
