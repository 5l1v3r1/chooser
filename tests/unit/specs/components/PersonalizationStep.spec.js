import { mount, createLocalVue } from '@vue/test-utils'
import PersonalizationStep from '@/components/PersonalizationStep'
import Buefy from 'buefy'
import LicenseUtilities from '@/utils/license-utilities'

describe('PersonalizationStep.vue', () => {
    let wrapper

    // Always creates a shallow instance of component
    beforeEach(() => {
        const localVue = createLocalVue()
        localVue.use(Buefy)
        localVue.use(LicenseUtilities)

        wrapper = mount(PersonalizationStep, {
            localVue,
            propsData: {
                value: {
                    authorName: '',
                    workTitle: '',
                    workUrl: '',
                    sourceWorkUrl: ''
                }
            }
        })
    })

    // Snapshot tests
    it('has the expected UI', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
