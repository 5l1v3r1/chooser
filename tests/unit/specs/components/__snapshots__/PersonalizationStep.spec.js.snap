// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalizationStep.vue has the expected UI 1`] = `
<section class="pers-form">
  <div class="field is-floating-in-label"><label class="label">Work Author</label>
    <div class="control is-clearfix"><input type="text" autocomplete="on" placeholder="John Silver" class="input">
      <!---->
      <!---->
      <!---->
    </div>
    <!---->
  </div>
  <div class="field is-floating-in-label"><label class="label">Title of Work</label>
    <div class="control is-clearfix"><input type="text" autocomplete="on" placeholder="This work" class="input">
      <!---->
      <!---->
      <!---->
    </div>
    <!---->
  </div>
  <div class="field is-floating-in-label"><label class="label">Work URL</label>
    <div class="control is-clearfix"><input type="text" autocomplete="on" placeholder="https://www.example.com/image.jpg" class="input">
      <!---->
      <!---->
      <!---->
    </div>
    <!---->
  </div>
  <div class="field is-floating-in-label"><label class="label">URL of the Source Work</label>
    <div class="control is-clearfix"><input type="text" autocomplete="on" placeholder="https://www.example.com/source.jpg" class="input">
      <!---->
      <!---->
      <!---->
    </div>
    <!---->
  </div>
</section>
`;
